karapace/protobuf/io.py

```diff
@@ -0,0 +1,228 @@
+# limitations under the License.
+import subprocess
+
+ZERO_BYTE = b'\x00'
+
+logger = logging.getLogger(__name__)
+
+
+def calculate_class_name(name: str) -> str:
+    return "c_" + hashlib.md5(name.encode('utf-8')).hexdigest()
+
+
+def check_props(schema_one, schema_two, prop_list):
+    try:
+        return all(getattr(schema_one, prop) == getattr(schema_two, prop) for prop in prop_list)
+    except AttributeError:
+        return False
+
+
+def match_schemas(writer_schema: ProtobufSchema, reader_schema: ProtobufSchema) -> bool:
+    # TODO (serge): schema comparison by fields required
+
+    return str(writer_schema) == str(reader_schema)
+
+
+def read_varint(bio: BytesIO) -> int:
+    """Read a variable-length integer.
+
+    :returns: Integer
+    """
+    varint = 0
+    read_bytes = 0
+
+    while True:
+        char = bio.read(1)
+        if len(char) == 0:
+            if read_bytes == 0:
+                return 0
+            raise EOFError(f"EOF while reading varint, value is {varint} so far")
+
+        byte = ord(char)
+        varint += (byte & 0x7F) << (7 * read_bytes)
+
+        read_bytes += 1
+
+        if not byte & 0x80:
+            return varint
+
+
+def read_indexes(bio: BytesIO):
+    try:
+        size: int = read_varint(bio)
+    except EOFError:
+        # TODO: change exception
+        raise IllegalArgumentException("problem with reading binary data")
+    if size == 0:
+        return [0]
+    return [read_varint(bio) for _ in range(size)]
+
+
+def find_message_name(schema: ProtobufSchema, indexes: List[int]) -> str:
+    result: list = []
+    types = schema.proto_file_element.types
+    for index in indexes:
+        try:
+            message = types[index]
+        except IndexError:
+            raise IllegalArgumentException(f"Invalid message indexes: {indexes}")
+
+        if message and isinstance(message, MessageElement):
+            result.append(message.name)
+            types = message.nested_types
+        else:
+            raise IllegalArgumentException(f"Invalid message indexes: {indexes}")
+
+    # for java we also need package name. But in case we will use protoc
+    # for compiling to python we can ignore it at all
+    return ".".join(result)
+
+
+def get_protobuf_class_instance(schema: ProtobufSchema, class_name: str) -> Any:
+    directory = config.DEFAULTS["protobuf_runtime_directory"]
+    proto_name = calculate_class_name(str(schema))
+    proto_path = f"{directory}/{proto_name}.proto"
+    class_path = f"{directory}/{proto_name}_pb2.py"
+    if not os.path.isfile(proto_path):
+        with open(f"{directory}/{proto_name}.proto", "w") as proto_text:
+            proto_text.write(str(schema))
+            proto_text.close()
+
+    if not os.path.isfile(class_path):
+        complete = subprocess.run([
+            "protoc",
+            "--python_out=./",
+            proto_path,
+        ], check=True)
+        if complete.returncode != 0:
```

Comment:
This is unnecessary. Because of the `check=True`, if the process failed `run` would have raised.

Reply:
ok